src/util/output.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Output helpers.
+
+/// Gets a colorized message for outputting to a particular stream. Do not use colors if the stream
+/// is not a tty.
+#[macro_export]
+macro_rules! colorize {
+    ($color:ident, $stream:ident, $arg:tt) => {{
+        let style = if atty::is(atty::Stream::$stream) {
+            ansi_term::Color::$color.normal()
+        } else {
+            ansi_term::Style::default()
+        };
+        style.paint($arg)
+    }};
+    ($color:ident, $stream:ident, $($arg:tt)*) => {{
+        let style = if atty::is(atty::Stream::$stream) {
+            ansi_term::Color::$color.normal()
+        } else {
+            ansi_term::Style::default()
+        };
+        style.paint(format!($($arg)*))
+    }};
+}
+
+/// Wrapper around `print!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! print_colored {
+    ($color:ident, $($arg:tt)*) => {
+        print!("{}", $crate::colorize!($color, Stdout, $($arg)*))
+    };
+}
+
+/// Wrapper around `println!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! println_colored {
+    ($color:ident, $($arg:tt)*) => {
+        println!("{}", $crate::colorize!($color, Stdout, $($arg)*))
+    };
+}
+
+/// Wrapper around `eprint!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! eprint_colored {
+    ($color:ident, $($arg:tt)*) => {
+        eprint!("{}", $crate::colorize!($color, Stderr, $($arg)*))
+    };
+}
+
+/// Wrapper around `eprintln!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! eprintln_colored {
+    ($color:ident, $($arg:tt)*) => {
+        eprintln!("{}", $crate::colorize!($color, Stderr, $($arg)*))
+    };
+}
+
+// COLORS:
+// The output colour of a particular object of text should convey some information about what
+// exactly it is. These are not hard and fast rules, but generally speaking:
+// - Blue for general information.
+// - Green for success.
+// - Red for errors and warnings.
+// - Cyan should be used for section titles.
+// - Yellow should represent field names -- i.e. UUIDs, versions, etc.
+//
+// By sticking to this guideline we ensure that the CLI stays consistent.
+
+/// Write an info message to stdout.
+#[macro_export]
+macro_rules! info {
+    ($($arg:tt)*) => {
+        $crate::print_colored!(Blue, "[INFO] ");
+        println!($($arg)*);
+    };
+}
```

Comment:
I think we should only keep one version of the macros, choosing only to emit logs to stdout or stderr. I would say that here we can simply log everything to stderr. If we want to have the choice, it might be best to have a configuration option to choose between stdout or stderr but for simplicity I think it's best to just log everything to stderr.

Reply:
On that note, maybe we can delete those macros and directly use the one from `log` with the `env_logger` implementor? The `verbosity` flag could set the log level and I am pretty sure that this crate handles colouring as well 💯 It's also customizable to make it look nice for the command line.

Reply:
Okay, I'll get rid of the stdout stuff!

I think using the `env_logger` module is a good idea, but I think I'll implement this in a future commit if that's alright with you? It might require some further change throughout this repo. 😛

Reply:
Sure that's fine!